Path-Check/gaen-mobile
# Add UI for opting into analytics sharing

### Description: 

Why:
As a product team, we want to collect useful analytics data from users
in a trustworthy and privacy-focused way.

This commit:
Adds UI to the settings screen to allow users to opt into anonymized
data sharing. Before they opt in, they will have a chance to review how
we will use their data and view their HA's privacy policy.

This commit introduces:
- An AnonymizedDataConsentScreen, which displays information about how
  we will use their analytics data
- A ShareAnonymizedDataSwitch, which allows users to turn data sharing
  on or off
- A SwitchListItem, which establishes a pattern for switch settings
  which we will use going forward.


#### Linked issues:

[GAEN-264](https://pathcheck.atlassian.net/jira/software/projects/GAEN/boards/33/?selectedIssue=GAEN-264)

#### Screenshots:

![toggly boi](https://user-images.githubusercontent.com/21161427/93387494-515a8d00-f837-11ea-84cc-60edd85884ac.gif)


#### How to test:

- Visit the settings screen
- You should see an analytics consent toggle that is switched off by default
- Switch it on
- You should be brought to a consent screen
- Click to consent
- You should be brought back to the settings screen, with the toggle switched on
